final/src/components/Exchange/Exchange.js

```diff
@@ -0,0 +1,160 @@
+    return getExchangeData(initial);
+  }
+
+  render() {
+    const { 
+      handleSubmit, 
+      formValues, 
+      submitting, 
+      data,
+      title,
+    } = this.props;
+
+    const fromTime = formValues ? new Date(formValues.from) : new Date();
+    const toTime = formValues ? new Date(formValues.to) : new Date();
+
+    const maxToTime = toTime;
+    maxToTime.setDate(maxToTime.getDate() - 1);
+
+    const maxFromTime = fromTime;
+    maxFromTime.setDate(maxFromTime.getDate() + 1);
+
+
+    return (
+      <React.Fragment>
+        <Form className="exchange-form" onSubmit={ handleSubmit(() => this.onSubmit()) }>
+          <section className="exchange-form__title-section">
+            <label>
+              Choose Title For Chart
+            </label>
+            <Field
+              name='title'
+              type='string'
+              autoComplete='title'
+              component={ InputField }
+              placeholder='Enter Line Chart Title'
+            />
+          </section>
+          <section>
+            <label>
+              Choose currency
+            </label>
+            <Field
+              name='currency'
+              type='number'
+              component={ DropdownField }
+              option={ currencyExchange }
+              placeholder='Select Currency'
+            />
+          </section>
+          <section className='exchange-form__from-section'>
+            <label>
+              From
+            </label>
+            <Field
+              name='from'
+              type='string'
+              component={ DatePickerField }
+              dateFormat={ DATE_FORMAT }
+              maxDate={ maxToTime }
+            />
+          </section>
+          <section className='exchange-form__to-section'>
+            <label>
+              To
+            </label>
+            <Field
+              name='to'
+              type='string'
+              component={ DatePickerField }
+              dateFormat={ DATE_FORMAT }
+              maxDate={ new Date() }
+              minDate={ maxFromTime }
+            />
+          </section>
+          <Button
+            content='Update'
+            disabled={ submitting }
+            loading={ submitting }
+          />
+        </Form>
+        <BarChart 
+          title={ title }
+          data={ data }
+        />
+      </React.Fragment>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  formValues: getFormValues('exchangeForm')(state),
+  formErrors: getFormSyncErrors('exchangeForm')(state),
+  data: state.exchange.data,
+  title: state.exchange.title,
+  initialValues: {
+    from: initial.from,
+    to: initial.to,
+    title: initial.title,
+    currency: initial.currency,
+  },
+});
+
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+  getExchangeData,
+  updateExchangeData,
+}, dispatch);
+
+const formConfig = {
+  form: 'exchangeForm',
+};
+
+const propTypes = {
+  formValues: PropTypes.object,
```

Comment:
It's not clear why all values are not required. Probably some of them should be.